### PR TITLE
feat(ISV-5129): add script to create product-level SBOMs

### PR DESCRIPTION
**Pull request by jedinym**
Add a python script to create product-level SBOMs in the `create-product-sbom` tekton task.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -68,4 +68,8 @@ LABEL com.redhat.component="release-service-utils"
 
 # Set HOME variable to something else than `/` to avoid 'permission denied' problems when writing files.
 ENV HOME=/tekton/home
-ENV PATH="$PATH:/home/pyxis:/home/utils:/home/pubtools-pulp-wrapper:/home/developer-portal-wrapper"
+ENV PATH="$PATH:/home/pyxis"
+ENV PATH="$PATH:/home/utils"
+ENV PATH="$PATH:/home/pubtools-pulp-wrapper"
+ENV PATH="$PATH:/home/developer-portal-wrapper"
+ENV PATH="$PATH:/home/sbom"
```
Comment (mmalina) [inline Dockerfile:75]: You will also need to add a `COPY` line for this above.

File: sbom/create-product-sbom.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
+# /usr/bin/env python3
```
Comment (mmalina) [inline sbom/create-product-sbom.py:1]: Typo: should be `#!/usr/bin/env python3`

**Comment by jedinym** [comment]
@mmalina thanks for the checks! are you also the maintainer of this repo?

**Comment by mmalina** [comment]
> @mmalina thanks for the checks! are you also the maintainer of this repo?

You're welcome. Yes I am. I haven't reviewed the Python code yet, I will do that later today hopefully.

**Review by mmalina — COMMENTED** [review]
The Python file looks good, but you will also need to add tests.

**Comment by jedinym** [comment]
@mmalina There are Tekton task tests for this one in the release-service PR. Should I add separate tests here?

**Comment by mmalina** [comment]
> @mmalina There are Tekton task tests for this one in the release-service PR. Should I add separate tests here?

I didn't realize you already had a test for this in https://github.com/konflux-ci/release-service-catalog/pull/556 . But yeah, it makes sense to have normal python unit tests for this here. And then typically in the tekton task tests we would mock such scripts and only check that they were called as expected. In this case since it's just manipulating data and it's doable to just let it run (compared to when a script calls services like Pyxis), it's probably ok not to mock the script. I would be inclined to do the mocking anyway, but I don't have a strong opinion on this. In any case, I would like to ask for the unit tests here. Sorry about that.

**Comment by jedinym** [comment]
@mmalina tests added, could you please take a look?

**Review by johnbieren — APPROVED** [review]
/ok-to-test
Can you rebase? I am good with this. I will give Martin til tomorrow to ask for any changes but then we can merge (no promises you won't have to rebase again though)

**Review by mmalina — APPROVED** [review]
@jedinym sorry for the delay. Can you rebase please? Then we can merge this.

**Comment by johnbieren** [comment]
Not sure why the last check still says queued pending approval. It does need rebase, but still

**Comment by jedinym** [comment]
@mmalina @johnbieren I had to do some small changes, could you please approve?

**Comment by mmalina** [comment]
> @mmalina @johnbieren I had to do some small changes, could you please approve?

What did you change? Python tests are failing. Also, most commits are not signed. Consider squashing the commits. They will be squashed on merge, but still.

**Review by johnbieren — APPROVED** [review]
It needs another rebase. I'd also recommend squashing the commits as @mmalina said

**Comment by jedinym** [comment]
Squashed and rebased.

**Pull request merged**